src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -756,7 +756,7 @@ def load_arguments(self, _):
         c.argument('sku', options_list=['--sku', '-s'], help='image sku')
         c.argument('publisher', options_list=['--publisher', '-p'], help='image publisher')
         c.argument('os_type', arg_type=get_enum_type(['Windows', 'Linux']), help='the type of the OS that is included in the disk if creating a VM from user-image or a specialized VHD')
-        c.ignore('os_state')  # service is not ready
+        c.argument('os_state', arg_type=get_enum_type(self.get_models('OperatingSystemStateTypes')), help="This property allows the user to specify whether the virtual machines created under this image are 'Generalized' or 'Specialized'.")
```

Comment:
Is there any link about this argument? What's the meaning of "Generalized" and "Specialized"?

Reply:
https://msazure.visualstudio.com/One/_git/Compute-CPlat-PIR?path=%2Fsrc%2FCAPS%2FDoc%2FPrivateGalleryDesignAndApiDoc.md&version=GBmaster

Reply:
I copy the help text from swagger.